### PR TITLE
Debian packaging: fix lintian tags; new package description

### PR DESCRIPTION
**Pull request by paride**
Fixes: LP: #1853376

Changes:

 * d/control: bump Standards-Version to 4.4.1 (no changes needed)

 * override package-supports-alternative-init-but-no-init.d-script
   Systemd is the only supported init system on Ubuntu/Debian.

 * d/cloud-init.preinst: remove support for ancient versions

 * debian: add upstream metadata file
   Fixes lintian tag: upstream-metadata-file-is-missing

 * debian: add i18n support to the debconf template files
   Fixes lintian tags:
    - untranslatable-debconf-templates
    - not-using-po-debconf

 * d/control: switch from git:// to https:// in the Vcs-Git field

 * d/cloud-init.postinst: guard the udevadm call
   Fixes lintian tag: udevadm-called-without-guard

 * d/rules: do not use dpkg-parsechangelog to get DEB_VERSION
   Fixes lintian tag: debian-rules-parses-dpkg-parsechangelog

 * d/control: rewrite the package description

 * d/control: add Rules-Requires-Root: no
   Fixes lintian tag: rules-requires-root-missing

 * d/control: add a Homepage field (https://cloud-init.io/)
   Fixes lintian tag: no-homepage-field

 * debian maint scripts: replace the '-e' in the shebang with a 'set -e'
   Fixes lintian tag: maintainer-script-without-set-e

 * d/cloud-init.postinst use 'command -v' to check if grub-install exists
   Fixes lintian tag: command-with-path-in-maintainer-script

 * d/changelog: remove trailing whitespaces
   Fixes lintian tag: file-contains-trailing-whitespace

### CODE REVIEW

**Comment by paride** [comment]
Left out are:

 - package-uses-old-debhelper-compat-version: I think we need to keep this as is for the SRUs
 - debian-watch-does-not-check-gpg-signature: we are not signing it
 - binary-without-manpage: I didn't intend to write the manpages now
 - systemd-service-file-missing-documentation-key: same as above
 - systemd-service-file-shutdown-problems: not to be fixed in the packaging branch; see https://lintian.debian.org/tags/systemd-service-file-shutdown-problems.html for more info

We may decide we don't like or want all of the changes I've made; they are split in individual commits and so easy to drop.

**Comment by paride** [comment]
The maintainer scripts are using a mixed tabs/spaces indentation. I didn't fix it here to avoid making the diff unreadable.

File: debian/upstream/metadata [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Name: cloud-init
+Contact: Josh Powers <josh.powers@canonical.com>
```
Comment (OddBloke) [inline debian/upstream/metadata]: Should we make this a mailing list?

Comment (paride) [reply debian/upstream/metadata]: I think it's a good idea; is there one more appropriate than ubuntu-server@lists.ubuntu.com?

**Pull request merged**